### PR TITLE
Implement read only flag

### PR DESCRIPTION
**Pull request by jreidinger**
## Problem

There is no indication if btrfs volume will be read only or not.

## Solution

Implement it.

## Testing

- *Tested manually*

## Screenshots
Note: No suitable icon in materials not found ( at least I do not like any of them ).

read_only_volume

### CODE REVIEW

File: web/src/components/storage/ProposalVolumes.jsx [hunk 1]
```diff
@@ -201,6 +201,10 @@ const VolumeRow = ({ columns, volume, options, isLoading, onEdit, onDelete }) =>
 
   const Details = ({ volume, options }) => {
     const hasSnapshots = volume.fsType === "Btrfs" && volume.snapshots;
+    const readOnly = volume.fsType === "Btrfs" && volume.readOnly;
+    console.log("details:");
+    console.log(volume);
+    console.log(options);
```
Comment (dgdavid) [inline web/src/components/storage/ProposalVolumes.jsx:207]: Leftovers, right?

File: web/src/components/storage/ProposalVolumes.jsx [hunk 2]
```diff
@@ -214,6 +218,8 @@ const VolumeRow = ({ columns, volume, options, isLoading, onEdit, onDelete }) =>
         <If condition={options.encryption} then={<Em icon={lockIcon}>{_("encrypted")}</Em>} />
         {/* TRANSLATORS: filesystem flag, it allows creating snapshots */}
         <If condition={hasSnapshots} then={<Em icon={snapshotsIcon}>{_("with snapshots")}</Em>} />
+        {/* TRANSLATORS: filesystem flag, system is read only */}
```
Comment (dgdavid) [inline web/src/components/storage/ProposalVolumes.jsx:221]: Serious question. System or filesystem?

**Review by dgdavid — COMMENTED** [review]
So far, LGTM. But have a look to these `console.log`. I think they are no really needed :)

**Comment by dgdavid** [comment]
> Tested manually

@jreidinger I have added few unit tests for the UI part. Hope you don't mind. Please, have a look.

File: web/src/components/storage/ProposalVolumes.jsx (hunk 1 above)
Comment (jreidinger) [reply web/src/components/storage/ProposalVolumes.jsx:207]: yeah, forget last force push

File: web/src/components/storage/ProposalVolumes.jsx (hunk 2 above)
Comment (jreidinger) [reply web/src/components/storage/ProposalVolumes.jsx:221]: good question. As it is generic volume, it is filesystem as system can live elsewhere.

**Review by dgdavid — APPROVED** [review]
LGTM

**Review by dgdavid — CHANGES_REQUESTED** [review]
Sorry, I think this one deserve an entry in the changelog files

**Comment by ancorgs** [comment]
I have concerns regarding the wording. This is not exactly a "read-only" filesystem.

I will elaborate (entering a meeting now)

**Comment by jreidinger** [comment]
@ancorgs OK, I will wait.

**Comment by ancorgs** [comment]
`volume.btrfs.read_only?` does not really mean the file-system is read-only.

To be precise, it means the root subvolume of the Btrfs file-system is read-only. But that does not apply to the whole file-system. Other subvolumes on the same Btrfs file-system (like /home, /opt, /srv or /var) are still writable.

If I see a Volume with an attribute `ReadOnly` set to true or if a see a `read-only` label in the table of volumes in the UI, I would expect the whole volume/filesystem to be read-only. Maybe we even have an use-case for that in the future.

I don't know how to name the D-Bus property or the label in the UI, but I don't like the current ReadOnly and read-only. Maybe we should try with "transactional".

**Comment by jreidinger** [comment]
@ancorgs  OK, makes sense. So lets use transactional name instead. I am just not sure if it btrfs exclusive

**Review by ancorgs — APPROVED** [review]
LGTM

**Comment by jreidinger** [comment]
@dgdavid Can you please also recheck? as it looks like your requests to change blocks merge

File: web/src/components/storage/ProposalVolumes.jsx [hunk 3]
```diff
@@ -214,6 +215,8 @@ const VolumeRow = ({ columns, volume, options, isLoading, onEdit, onDelete }) =>
         <If condition={options.encryption} then={<Em icon={lockIcon}>{_("encrypted")}</Em>} />
         {/* TRANSLATORS: filesystem flag, it allows creating snapshots */}
         <If condition={hasSnapshots} then={<Em icon={snapshotsIcon}>{_("with snapshots")}</Em>} />
+        {/* TRANSLATORS: filesystem flag, filesystem is read only */}
```
Comment (dgdavid) [inline web/src/components/storage/ProposalVolumes.jsx:218]:
```suggestion
        {/* TRANSLATORS: flag for transactional file system  */}
```

**Review by dgdavid — APPROVED** [review]
LGTM, just check the comment for translators

**Pull request merged**